### PR TITLE
Issue 152 fixing tests

### PR DESCRIPTION
**Pull request by dieppa**

### CODE REVIEW

File: dimmer-core/src/main/java/com/github/cloudyrock/dimmer/logic/DimmerAspect.java [hunk 1]
```diff
@@ -16,7 +16,7 @@
  * @see DimmerFeature
  */
 @Aspect
-class DimmerAspect {
+public class DimmerAspect {
```
Comment (dhivesh) [inline dimmer-core/src/main/java/com/github/cloudyrock/dimmer/logic/DimmerAspect.java]: I know you don’t like this, but it had to be done :)

File: dimmer-core/src/main/java/com/github/cloudyrock/dimmer/logic/DimmerBuilder.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package com.github.cloudyrock.dimmer.builder;
+package com.github.cloudyrock.dimmer.logic;
```
Comment (dhivesh) [inline dimmer-core/src/main/java/com/github/cloudyrock/dimmer/logic/DimmerBuilder.java]: I understand why you added a package named logic, but there is prob a better name we can figure out in the future

File: dimmer-core/src/main/java/com/github/cloudyrock/dimmer/logic/FeatureExecutorImpl.java [hunk 3]
```diff
@@ -54,22 +52,33 @@ public Object executeDimmerFeature(String feature,
...
     }
 
     Object executeFeature(String feature, String operation, FeatureInvocation featureInvocation) {
-        final BehaviourKey key = new BehaviourKey(feature, operation);
+        final Behaviour.BehaviourKey key = new Behaviour.BehaviourKey(feature, operation);
         final Object result = behaviours.get(key).apply(featureInvocation);
-        checkReturnType(featureInvocation.getReturnType(), result);
+        if (!isRightReturnedType(featureInvocation.getReturnType(), result)) {
+
+//            final String EXCEPTION_MISMATCHED_RETURNED_TYPE2 =
```
Comment (dhivesh) [inline dimmer-core/src/main/java/com/github/cloudyrock/dimmer/logic/FeatureExecutorImpl.java]: Delete comments

File: dimmer-core/src/test/java/com/github/cloudyrock/dimmer/DimmerBuilderBehaviourIT.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
+package com.github.cloudyrock.dimmer;
```
Comment (dhivesh) [inline dimmer-core/src/test/java/com/github/cloudyrock/dimmer/DimmerBuilderBehaviourIT.java]: +1

File: dimmer-core/src/test/java/com/github/cloudyrock/dimmer/util/BuilderTestUtil.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
+package com.github.cloudyrock.dimmer.util;
```
Comment (dhivesh) [inline dimmer-core/src/test/java/com/github/cloudyrock/dimmer/util/BuilderTestUtil.java]: +1

File: dimmer-core/src/main/java/com/github/cloudyrock/dimmer/logic/DimmerBuilder.java (hunk 2 above)
Comment (dieppa) [reply dimmer-core/src/main/java/com/github/cloudyrock/dimmer/logic/DimmerBuilder.java]: 100%

**Pull request merged**